core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -204,12 +209,17 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
         stats.indexEntriesRefreshed.increment(count)
         logger.info(s"Refreshed downsample index with new records numRecords=$count " +
           s"dataset=$rawDatasetRef shard=$shardNum fromHour=$fromHour toHour=$toHour")
+        partKeyIndex.commit()
```

Comment:
I am not sure ``commit`` guarantees sync to file system. We will have to look into Lucene internals.

Reply:
EDIT: I think commit looks ok at least from the documentation 


https://lucene.apache.org/core/7_4_0/core/org/apache/lucene/index/IndexWriter.html#commit--